### PR TITLE
[W4.3d]&[W4.4c][T09-A2] Lim Yong Song

### PR DESCRIPTION
**Pull request by limyongsong**
Created Classes for Block, Unit, PostalCode and Street that only hold values.
Address Class itself is not fundamentally changed.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -22,13 +27,27 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+        //Added for week4LO
+        String[] addressHolder = new String[99];
+        int i=0;
+       for(String retVal: trimmedAddress.split(",")){
```
Comment (limyongsong) [inline src/seedu/addressbook/data/person/Address.java]: Split loop is done with reference to https://www.tutorialspoint.com/java/java_string_split.html

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -12,6 +12,11 @@
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
+    Block block;
+    Street street;
+    Unit unit;
+    PostalCode postalCode;
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java]: Don't forget to specify the access type. E.g. `public`,` private` and so on.

File: src/seedu/addressbook/data/person/Unit.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package seedu.addressbook.data.person;
+
+public class Unit {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Unit.java]: No header comment.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -22,13 +27,27 @@
...
+        postalCode = new PostalCode(addressHolder[3]);
+
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
     }
-
+    public String getBlock(){ return block.getBlock();}
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java]: Coding standard violations.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -22,13 +27,27 @@
...
+        //Added for week4LO
+        String[] addressHolder = new String[99];
+        int i=0;
+       for(String retVal: trimmedAddress.split(",")){
+            addressHolder[i++] = retVal;
+        }
+        block = new Block(addressHolder[0]);
+        street = new Street(addressHolder[1]);
+        unit = new Unit(addressHolder[2]);
+        postalCode = new PostalCode(addressHolder[3]);
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Address.java]: Magic number??

File: doc/UserGuide.md [hunk 6]
```diff
@@ -46,7 +46,7 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
 > Persons can have any number of tags (including 0)
 
 Examples: 
-* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
+* `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01, S123456`
```
Comment (jeffryhartanto) [inline doc/UserGuide.md]: Good job for updating the UserGuide.

**Comment by jeffryhartanto** [comment]
@limyongsong Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**